torch_geometric/nn/models/mlp.py

```diff
@@ -114,11 +124,15 @@ def __init__(
         self.norms = torch.nn.ModuleList()
         iterator = channel_list[1:-1] if plain_last else channel_list[1:]
         for hidden_channels in iterator:
-            if batch_norm:
-                norm = BatchNorm1d(hidden_channels, **batch_norm_kwargs)
+            if norm is not None:
+                norm_layer = normalization_resolver(
+                    norm,
+                    hidden_channels,
+                    **(norm_kwargs or {}),
+                )
             else:
-                norm = Identity()
-            self.norms.append(norm)
+                norm_layer = Identity()
```

Comment:
do you need to do this? could we just have no norm layer in the case that none are needed rather than adding the identity?

Reply:
This shifts the efforts of checking this to the `forward` call. I think either way is fine.